app/components/drawer_layout_polyfill.js

```diff
@@ -0,0 +1,401 @@
+import React, { Component } from 'react';
+import {
+    Animated,
+    Dimensions,
+    Keyboard,
+    PanResponder,
+    StyleSheet,
+    TouchableWithoutFeedback,
+    View,
+    I18nManager,
+} from 'react-native';
+
+const MIN_SWIPE_DISTANCE = 3;
+const DEVICE_WIDTH = parseFloat(Dimensions.get('window').width);
+const THRESHOLD = DEVICE_WIDTH / 2;
+const VX_MAX = 0.1;
+
+const IDLE = 'Idle';
+const DRAGGING = 'Dragging';
+const SETTLING = 'Settling';
+
+export default class DrawerLayout extends Component {
+    /*static propTypes = {
+        children: any,
+        drawerBackgroundColor?: string,
+        drawerLockMode?: 'unlocked' | 'locked-closed' | 'locked-open',
+        drawerPosition: 'left' | 'right',
+        drawerWidth: number,
+        keyboardDismissMode?: 'none' | 'on-drag',
+        onDrawerClose?: Function,
+        onDrawerOpen?: Function,
+        onDrawerSlide?: Function,
+        onDrawerStateChanged?: Function,
+        renderNavigationView: () => any,
+        statusBarBackgroundColor?: string,
+        useNativeAnimations?: boolean,
+    };*/
+    
+    static defaultProps = {
+        drawerWidth: 0,
+        drawerPosition: 'left',
+        useNativeAnimations: false,
+    };
+
+    static positions = {
+        Left: 'left',
+        Right: 'right',
+    };
+
+    constructor(props, context) {
+        super(props, context);
+
+        this.state = {
+            accessibilityViewIsModal: false,
+            drawerShown: false,
+            openValue: new Animated.Value(0),
+        };
+    }
+
+    getDrawerPosition() {
+        const { drawerPosition } = this.props;
+        const rtl = I18nManager.isRTL;
```

Comment:
we don't have RTL support, is this code taken from somewhere? why aren't we giving credit?

Reply:
yes, this is from library, I pulled it because we need to do some iOS fixes like DEVICE_WIDTH etc.